geoutils/raster/satimg.py

```diff
@@ -428,8 +428,8 @@ def __parse_metadata_from_file(self, fn_meta: str | None) -> None:
 
         return None
 
-    def copy(self, new_array: NDArrayNum | None = None) -> SatelliteImage:
-        new_satimg = super().copy(new_array=new_array)  # type: ignore
+    def copy(self, new_array: NDArrayNum | None = None, cast_nodata: bool = False) -> SatelliteImage:
```

Comment:
why is the default False, whereas it was True for the Raster class?

Reply:
good catch! fixed